src/net.cpp

```diff
@@ -417,6 +417,9 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
     // Resolve
     const uint16_t default_port{pszDest != nullptr ? GetDefaultPort(pszDest) :
                                                      m_params.GetDefaultPort()};
+
+    // Collection of addresses to try to connect to: either all dns resolved addresses (if pszDest is provided) or addrConnect otherwise.
+    std::vector<CAddress> connect_to{};
```

Comment:
nit: do we still need `Shuffle(resolved.begin(), resolved.end(), FastRandomContext());`? It certainly doesn't achieve the same goal as before... I'd suggest either dropping it altogether, or document this new purpose (either inline or in the commit message).

Reply:
What was the goal before? If it was for some load balancing, e.g. if it returns 3 addresses to which we can connect to not always connect to the first result? (or have two different nodes connect to the first result)

If "yes" then that still holds with this PR?

Reply:
This was added by me as a suggestion from @vasild in https://github.com/bitcoin/bitcoin/pull/28155#discussion_r1296832854 to maintain the older logic, but it comes all the way from https://github.com/bitcoin/bitcoin/commit/6387f397b323b0fb4ca303fe418550f5465147c6#diff-00021eed586a482abdb09d6cdada1d90115abe988a91421851960e26658bed02R430.

There is no comment nor commit message motivating it though.

I think randomizing the order in which the resolved addresses are tried wouldn't hurt though.

Reply:
Yeah, after looking more into the current PR, it achieves nearly the same goal...
I'm fine with not doing anything here.

But it would be cool to understand better what it achieves :)

Here's what i think is going on. A DNS implementation may (or may not) return results in the same order, and then all the nodes will end up making connections with the same result. Then it's useful for load balancing, and to avoid DNS influencing the connectivity in general.
Might be worth adding this comment.

Reply:
I think we can go ahead with this PR without waiting for clarification.